tools/cli/README.md

```diff
@@ -0,0 +1,108 @@
+(`./cadence help`, `./cadence help [domain|workflow]` will also print help messages)
+
+**Note:** make sure you have cadence server running before using CLI 
+
+### Domain operation examples 
+- Register a new domain named "samples-domain":  
+```
+./cadence --domain samples-domain domain register 
+# OR using short alias  
+./cadence --do samples-domain domain re
+```   
+- View "samples-domain" details:   
+```
+./cadence --domain samples-domain domain describe  
+```
+
+**Tips:**  
+to avoid repeated input global option **domain**, user can export domain-name in environment variable CADENCE_CLI_DOMAIN.
+```
+export CADENCE_CLI_DOMAIN=samples-domain
+
+# then just run commands without --domain flag, like
+./cadence domain desc
+```
+
+### Workflow operation examples
+(The following examples assume you already export CADENCE_CLI_DOMAIN environment variable as Tips above)
+
+- Run workflow: Start a workflow and see it's progress, this command doesn't finish until workflow completes
+```
+./cadence workflow run --tl helloWorldGroup --wt main.Workflow --et 60 -i '"cadence"'
+
+# view help messages for workflow run
+./cadence workflow run -h
+``` 
+Brief explanation:  
+To run a workflow, user must specify 
+1. Tasklist name (--tl), 
+2. Workflow type (--wt), 
+3. Execution start to close timeout in seconds (--et), 
+
+Example uses [this cadence-samples workflow](https://github.com/samarabbas/cadence-samples/blob/master/cmd/samples/recipes/helloworld/helloworld_workflow.go) 
+and it takes a string as input, so there is the `-i '"cadence"'`. Single quote `''` is used to wrap input as json. 
+
+**Note:** you need to start worker so that workflow can make progress.  
+(Run `make && ./bin/helloworld -m worker` in cadence-samples to start the worker)
+
+- Start workflow: 
+```
+./cadence workflow start --tl helloWorldGroup --wt main.Workflow --et 60 --dt 10 -i '"cadence"'
+
+# view help messages for workflow start
+./cadence workflow start -h
+
+# for workflow with multiple input, seperate each json with space/newline like
+./cadence workflow start --tl helloWorldGroup --wt main.WorkflowWith3Args --et 60 --dt 10 -i '"your_input_string" 123 {"Name":"my-string", "Age":12345}'
+```
+Workflow `start` command is similar to `run` command and takes same flag options. But it just start the workflow and immediately return workflow_id and run_id.  
+User need to run `show` to view workflow history/progress.  
+
+- Show workflow history
+```
+./cadence workflow show -w 3ea6b242-b23c-4279-bb13-f215661b4717 -r 866ae14c-88cf-4f1e-980f-571e031d71b0
+
+# if run_id is not provided, it will show the latest run history for that workflow_id
+./cadence workflow show -w 3ea6b242-b23c-4279-bb13-f215661b4717
+```
+
+- List open or closed workflow executions
+```
+./cadence workflow list
+```
+
+- Query workflow execution
+```
+# use build-in query type "__stack_trace" which is supported by cadence client library
+./cadence workflow query -w <wid> -r <rid> 
```

Comment:
What about 
cadence workflow stack -w ...

Reply:
I think query at least link me with QueryWorkflow of cadence-client, no sure why you propose stack?

Reply:
Because if you don't specify query type you default to stack which is confusing. I would require query type, but provide separate command "stack" which would execute query type "__stack_trace" which is hard to remember.

Reply:
Added `stack` command